### DISCUSSION TITLE
Decoding optional field values to Go

### DISCUSSION THREAD

**Discussion by brandoshmando** [cue-lang/cue#1633] [Q&A] [answered]
When using the `Value.Decode` method to decode key/values into Go structs, optional fields are not populated.

As a simple example (using 0.4.2):
```golang
func main() {
	typ := struct {
		F1 string
	}{}
	input := `
	{
		F1?: "value"
	}`

	ctx := cuecontext.New()
	val := ctx.CompileString(input)

	val.Decode(&typ)
	fmt.Printf("%+v\n", typ)
	fmt.Println(val.LookupPath(cue.MakePath(cue.Str("F1").Optional())).IsConcrete())
}

// Output:
// {F1:}
// true
```

The expectation here is that since the value of optional field `F1?` is concrete, it should be decoded into the given struct. Couple of questions:
1) Is omitting optional fields when decoding to Go values expected behavior?
2) If so, is there any way around this using `Value.Decode` i.e. a way to extract values from optional fields into Go values, ideally without having to look each one up with `LookupPath`?

**Comment by eonpatapon** [comment, marked as answer]
I think this is a normal behavior because for `cue` in this example `F1` is not defined. It just says if `F1` is defined then it's value must be `"value"`.

Instead I would use a default value if you need the value to be decoded in the go struct:

```cue
{
    F1: *"value" | string
}
```

**Reply by brandoshmando** [reply]
Ahh yea looking at this again, that makes sense. I was misinterpreting concreteness with defined _schema_ field values. Thanks for the response!
